core/domain/rules_registry.py

```diff
@@ -24,16 +24,32 @@
 from core import constants
 from core import feconf
 
+from typing import Dict, List, Optional
+from typing_extensions import TypedDict
+
+
+class RuleSpecsExtensionDict(TypedDict):
+    """Dictionary representation of rule specs of an extension."""
+
+    interactionId: str
+    format: str
+    ruleTypes: Dict[str, Dict[str, List[str]]]
+
 
 class Registry:
     """Registry of rules."""
 
-    _state_schema_version_to_html_field_types_to_rule_specs = {}
+    _state_schema_version_to_html_field_types_to_rule_specs: Dict[
+        Optional[int],
```

Comment:
Why is this optional?

Reply:
Hmm, is `get_html_field_types_to_rule_specs` ever called with `None`?

Reply:
`get_html_field_types_to_rule_specs` is never called with None value explicitly, but it is called without any value which makes the `state_schema_version` None ( because the default value of state_schema_version is None ). 

Some of the places where it is called without any value:
https://github.com/oppia/oppia/blob/ba4f072e494fd59df53fecc37e67cea7f9727234/core/domain/state_domain.py#L923
https://github.com/oppia/oppia/blob/ba4f072e494fd59df53fecc37e67cea7f9727234/core/domain/state_domain.py#L213

Reply:
Hmm, I see. Looking at the context of this code it seems that `None` equals the latest version, can we maybe then do smth like `if None` then set to the latest schema version and use that?

Reply:
I think the latest version is `48` and I think `None` is used when we are fetching the data from `html_field_types_to_rule_specs.json` 
https://github.com/oppia/oppia/blob/6f447b0bbf68d0bcdfe75eb211ed2f28bda8bef1/core/domain/rules_registry.py#L62
and in `html_field_types_to_rule_specs.json` there is no data, apart from an empty dict. https://github.com/oppia/oppia/blob/6f447b0bbf68d0bcdfe75eb211ed2f28bda8bef1/extensions/interactions/html_field_types_to_rule_specs.json#L1

So, if we declare `state_schema_version` as `48` when it is none ( using ` if None` ), then maybe it create confusion among other developers. 

Can you give suggestions ( Also, please correct me if I'm wrong somewhere )